courses/views.py

```diff
@@ -144,13 +145,10 @@ def remove_student_from_course(request, course_id, student_id):
 
 
 def student_can_add_course(request, course_id, student_id):
-    # this function is implemented incorrectly
-    # it should check the student_registration_open
-    # on course model, noura, open a ticket and fix this
-    if 'student_registration_open' in request.GET:
+    if Course.objects.get(pk=course_id).student_registration_open:
         course = Course.objects.get(pk=course_id)
         student = Student.objects.get(pk=student_id)
-        student.courses.add(course)
+        course.students.remove(student)
```

Comment:
U are correct it should be add

Also, there needs to be an if statement
The student can only register if student_registration_open is true... It is a field on the student object